apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -199,6 +210,22 @@ public override CommandType CommandType
         /// </summary>
         public new SpannerParameterCollection Parameters { get; } = new SpannerParameterCollection();
 
+        /// <summary>
+        /// The keys of the rows that should be read from the target table if the command is Read, or null otherwise.
+        /// </summary>
+        public KeySet KeySet { get; }
+
+        /// <summary>
+        /// The index to use to search and order rows from the target table if the command is Read and the read
+        /// operation should use a different index than the primary key, or null otherwise.
+        /// </summary>
+        public string IndexName { get; }
+
+        /// <summary>
+        /// The maximum number of rows to read if the command is Read, or zero otherwise.
+        /// </summary>
+        public long Limit { get; }
+
```

Comment:
I'm on the fence about these, but wondering if they should go on `SpannerCommandTextBuilder` instead. The way I see it, KeySet is equivalent to ParameterCollection and the rest should just go as part of the "command text"?

Also regardless of wether they are here or there, and given that they are optional, I would create a class `ReadOptions` and add these. That way we can add new ones if the API team adds them withouth risking breaking changes on the methods that received them as parameters.

Reply:
I was a little bit back and forth on that as well. I've moved them into the `CommandTextBuilder` now, and I think that is cleaner and also consistent with what we keep in the `Command` and what we keep in the `CommandTextBuilder` for queries.
I also added a `ReadOptions` class.